src/openrct2/core/String.cpp

```diff
@@ -242,6 +242,26 @@ namespace String
         return false;
     }
 
+    bool Contains(std::string_view haystack, std::string_view needle, bool ignoreCase)
+    {
+        if (!ignoreCase)
+            return haystack.find(needle) != std::string_view::npos;
+
+        if (needle.size() > haystack.size())
+            return false;
```

Comment:
This could be moved to be the first early-out. `std::string_view::find` does not guarantee to have this same check inside.

Reply:
Surely `std::string_view::find` could never return true if `needle.size() > haystack.size()`?

Reply:
Exactly my point, so it's a minor optimization to move this before the previous `if` block. Since you already know it will not result in a valid iterator, you do not have to call `haystack.find` at all.

Reply:
ok